dex/msgjson/types.go

```diff
@@ -826,6 +840,71 @@ func (n *NotifyFee) Serialize() []byte {
 	return append(b, uint64Bytes(n.Time)...)
 }
 
+// Reinstate is the payload for the ReinstateRoute request.
+type Reinstate struct {
+	Signature
+	DEXPubKey    Bytes  `json:"dexpubkey"`
+	ClientPubKey Bytes  `json:"clientpubkey"`
+	AccountProof Bytes  `json:"accountproof"`
+	CoinID       Bytes  `json:"coinid"`
+	Time         uint64 `json:"timestamp"`
+}
+
+// Serialize serializes the Register data.
+func (r *Reinstate) Serialize() []byte {
+	// serialization: dexpubkey (33) + clientpubkey (33) +
+	// accountproof 96 + coinID (variable, ~32+2) + time (8) = 204
+	s := make([]byte, 0, 143)
+	s = append(s, r.DEXPubKey...)
+	s = append(s, r.ClientPubKey...)
+	s = append(s, []byte(r.AccountProof)...)
+	s = append(s, r.CoinID...)
+	return append(s, uint64Bytes(r.Time)...)
+}
+
+// Account proof is information necessary to prove that the DEX server accepted
+// the account's fee payment. The fee coin is not part of the proof, since it
+// is already stored as part of the AccountInfo blob.
+type AccountProof struct {
```

Comment:
Move this type back to **client/db**. This is not a **msgjson** type.

Reply:
With the current solution, AccountProof is needed by the dex server. msgjson was my first goto for object shared between client & server. If there is a more appropriate solution, please advise and I will change accordingly.